packages/zowe-explorer/src/dataset/actions.ts

```diff
@@ -701,36 +619,226 @@ async function handleUserSelection(newDSProperties, dsType): Promise<string> {
+    templates?.forEach((item) => {
+        templateNames.push(item.name);
+    });
+    return templateNames;
+}
+
+function compareDsProperties(type: string, datasetProvider: api.IZoweTree<api.IZoweDatasetTreeNode>): boolean {
+    let isMatch = true;
+    const templates: api.dsAlloc[] = datasetProvider.getDsTemplates();
+    let propertiesFromDsType: api.dsAlloc;
+    // Look for template
+    templates?.forEach((template) => {
+        if (type === template?.name) {
+            propertiesFromDsType = template;
+        }
+    });
+    if (!propertiesFromDsType) {
+        propertiesFromDsType = getDefaultDsTypeProperties(type);
+    }
+    newDSProperties?.forEach((property) => {
+        Object.keys(propertiesFromDsType).forEach((typeProperty) => {
+            if (typeProperty === property.key) {
+                if (property.value !== propertiesFromDsType[typeProperty].toString()) {
+                    isMatch = false;
+                    return;
+                }
+            }
+        });
+    });
+    return isMatch;
+}
+
+function getDsProperties(type: string, datasetProvider: api.IZoweTree<api.IZoweDatasetTreeNode>): api.dsAlloc {
+    const templates: api.dsAlloc[] = datasetProvider.getDsTemplates();
+    let propertiesFromDsType: api.dsAlloc;
+    // Look for template
+    templates?.forEach((template) => {
+        if (type === template?.name) {
+            if (template.dsorg === "PS") {
+                typeEnum = 4;
+            } else {
+                typeEnum = 3;
+            }
+            propertiesFromDsType = template;
         }
-        return Promise.resolve(handleUserSelection(newDSProperties, dsType));
+    });
+    if (!propertiesFromDsType) {
+        propertiesFromDsType = getDefaultDsTypeProperties(type);
     }
+    newDSProperties?.forEach((property) => {
+        Object.keys(propertiesFromDsType).forEach((typeProperty) => {
+            if (typeProperty === property.key) {
+                property.value = propertiesFromDsType[typeProperty].toString();
+                property.placeHolder = propertiesFromDsType[typeProperty];
+            }
+        });
+    });
+    return propertiesFromDsType;
+}
+
+function getDefaultDsTypeProperties(dsType: string): api.dsAlloc {
+    typeEnum = getDataSetTypeAndOptions(dsType)?.typeEnum;
+    const cliDefaultsKey = globals.CreateDataSetTypeWithKeysEnum[typeEnum]?.replace("DATA_SET_", "");
+    return zowe.CreateDefaults.DATA_SET[cliDefaultsKey] as api.dsAlloc;
+}
+
+async function allocateOrEditAttributes(): Promise<string> {
+    const stepThreeOptions: vscode.QuickPickOptions = {
+        ignoreFocusOut: true,
+        canPickMany: false,
+    };
+    const allocate = `\u002B ${localizedStrings.allocString}`;
+    const editAtts = `\u270F ${localizedStrings.editString}`;
+    const stepThreeChoices = [allocate, editAtts];
+    // eslint-disable-next-line no-return-await
+    return await api.Gui.showQuickPick(stepThreeChoices, stepThreeOptions);
```

Comment:
My above comment (regarding changing return type to `Thenable<string>`) could also apply here 😅